providers/common/provider_util.c

```diff
@@ -365,3 +366,8 @@ int ossl_prov_memdup(const void *src, size_t src_len,
     }
     return 1;
 }
+
+int ossl_param_is_empty(const OSSL_PARAM params[])
+{
+    return params == NULL || params->key == NULL;
+}
```

Comment:
Please, please make the function static inline in the header.

Reply:
Why?  Function calls are basically free on modern CPUs.  LTO would inline if required.
Let's not prematurely optimise things.  The OSSL_PARAM code isn't going to be on a critical path.

Reply:
LTO is not enabled by default. Also I would not call function calls free. Yeah, cheap, but free?

Reply:
> The OSSL_PARAM code isn't going to be on a critical path.

This is not true for symmetric algos. They are fairly critical path there.

Reply:
Moved to header